src/dashboard/src/media/js/backlog/backlog-search.js

```diff
@@ -4,34 +4,106 @@ $(document).ready(function()
 
     function render_file_actions_col(relative_path, type, row_data) {
       return '<a class="btn btn-default fa-download fa" target="_blank" href="' + '/filesystem/download_ss/?filepath=' +
-              Base64.encode('/originals/' + relative_path) + '">' + gettext('Download') + '</a>';
+              Base64.encode('/originals/' + relative_path) + '"> ' + gettext('Download') + '</a>';
     }
 
     function render_transfer_actions_col(uuid, type, row_data) {
-      return '<a class="btn btn-default" href="/backlog/download/' + uuid + '">' + gettext('Download') + '</a>' +
+      return '<a class="btn btn-default fa-download fa" href="/backlog/download/' + uuid + '"> ' + gettext('Download') + '</a>' +
              '<a href="/backlog/delete/' + uuid + '"><img title="' + gettext('Request deletion') + '" \
              class="delete-icon" src="/media/images/delete.png"></a>';
     }
 
+    function render_status(row_data) {
+      return row_data == true ? 'Delete requested' : 'Stored';
+    }
+ 
+    function get_state_url_params() {
+      return $('#id_show_files').prop('checked') ? 'transferfiles/' : 'transfers/';
+    }
+
+    function get_default_hidden_columns() {
+      return $('#id_show_files').prop('checked') ? [2, 3] : [3, 4, 6];
+    }
+
+    function get_action_column_index() {
+      return $('#id_show_files').prop('checked') ? [4] : [7];
+    }
```

Comment:
I'd be happy to! These are using JavaScript's [ternary operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator), basically a short-hand for conditional logic.

`get_default_hidden_columns` could be rewritten without the ternary operator as:

```
function get_default_hidden_columns() {
  if ( $('#id_show_files').prop('checked') ) {
    return [2, 3];
  } else {
    return [3, 4, 6];
  }  
}
```
Personally I find the ternary form easier to read, but that might just be due to familiarity. I've noticed it used elsewhere in the Angular frontend app as well, but I'm not sure if we have standard practice/style for JS in Archivematica?

If you mean the logic rather than the syntax, these functions are returning column indexes for the columns in the DataTable, for use in the DataTable initialization function. `get_default_hidden_columns` returns an array containing the indices for the columns which should be hidden by default if there's no saved saved (we decided to hide all of the new columns by default). These indices change depending on whether the "Show files?" button is checked (and hence the files table, rather than transfers table is being displayed). `get_action_column_index` is returning an array containing the column index of the "Actions" column for the table that's currently displayed (transfers or files), so that I can disable sorting on that column.

Maybe `get_default_hidden_columns` should be renamed to `get_default_hidden_columns_index`? I'd also be happy to write some in-line comments explaining what's going on.

Reply:
Ah, yep, that's increasingly familiar to me. 

I'm not going to suggest changing from ternary to long-hand here especially if it's elsewhere in our code. 

Borrowing from the longhand though: `return [3, 4, 6];` -- my question was more specifically about the indices here. Do we only have numbers to refer to these columns? Have we any tools in our kit to make these self-describing? Do we only have integers to refer to columns or do we have access to a column name at all? e.g. key/value. 

No worries at all if not.

Reply:
Per the [DataTable docs](https://datatables.net/reference/option/columnDefs.targets) you should be able to target a column by a string representing the corresponding <th> value in the table. In practice I found this doesn't work very reliably, especially when switching languages in the interface, whereas the integer values work consistently.